compiler/src/iree/compiler/Codegen/Common/GPU/VectorReductionToGPU.cpp

```diff
@@ -312,17 +316,16 @@ struct VectorReductionToGPUPass final
 
 private:
   bool expandSubgroupReduction;
-  std::function<int(mlir::FunctionOpInterface)> getWarpSize;
+  bool pickLargestSubroupSize;
```

Comment:
This flag is a bit weird, shouldn't the choice be defined by the pipeline we are using?

Reply:
Yeah the root issue is that for RDNA we have two modes, wave32 and wave64. If go through HIP wave32 is the default and if go through Vulkan, wave64 is the default.. it's a bit annoying. I think we should be able to mass convert vulkan side to use wave32. created https://github.com/iree-org/iree/issues/18037 to track.